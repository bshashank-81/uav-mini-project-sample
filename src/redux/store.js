import { combineReducers, createStore, compose } from "redux";
import listReducer from "../locationListPage-state/reducers/listReducer";

const reducer = combineReducers({
  locations: listReducer,
});

const store = createStore(
  reducer,
  process.env.NODE_ENV !== "production" &&
    typeof window === "object" &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION__({})
    : compose
);

export default store;
