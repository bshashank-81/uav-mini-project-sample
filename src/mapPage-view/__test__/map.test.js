import React from "react";
import "@testing-library/jest-dom/extend-expect";
import Map from "../components/map";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import store from "../../redux/store";
import { render, fireEvent, act } from "@testing-library/react";
import PublicIcon from "@material-ui/icons/Public";
import { createStore } from "redux";
import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

jest.mock("mapbox-gl", () => ({
  GeolocateControl: jest.fn(),
  Map: jest.fn(() => ({
    addControl: jest.fn(),
    on: jest.fn(),
    resize: jest.fn(),
  })),
  NavigationControl: jest.fn(),
  MapboxGeocoder: jest.fn(),
  Marker: jest.fn(() => ({
    setLngLat: jest.fn(() => ({
      addTo: jest.fn(),
    })),
  })),
}));

var value = {
  fake_access_token: true,
  token: {
    access_token: "fake_access_token",
  },
};

const handleViewMode = jest.fn(() => ({
  Map: jest.fn(() => ({
    setStyle: jest.fn(),
  })),
}));

describe("Map component", () => {
  Object.defineProperty(window, "localStorage", {
    value: {
      getItem: jest.fn().mockReturnValue(JSON.stringify(value)),
    },
  });
  const store = createStore(() => ({
    locations: [
      { id: 1, lat: 40.712776, lng: -74.005974 },
      { id: 2, lat: 42.360081, lng: -71.058884 },
    ],
  }));

  test("Matches the snapshot", () => {
    const maps = create(
      <Router>
        <Provider store={store}>
          <Map />
        </Provider>
      </Router>
    );
    expect(maps.toJSON()).toMatchSnapshot();
  });

  test("Map component should be there", () => {
    const { getByTestId } = render(
      <Router>
        <Provider store={store}>
          <Map />
        </Provider>
      </Router>
    );
    expect(getByTestId("map")).toBeInTheDocument();
  });

  test("public icon should be there", () => {
    const { getByTestId } = render(
      <Router>
        <Provider store={store}>
          <Map>
            <PublicIcon id="publicIcon" onClick={handleViewMode} />
          </Map>
        </Provider>
      </Router>
    );
    expect(getByTestId("publicIcon")).toBeInTheDocument();
    fireEvent.click(getByTestId("publicIcon"));
  });
});
