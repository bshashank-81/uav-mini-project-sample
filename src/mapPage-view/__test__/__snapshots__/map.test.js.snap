// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component Matches the snapshot 1`] = `
<div
  data-testid="map"
>
  <div
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <div
      className="info"
      id="info"
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div
      style={
        Object {
          "bottom": "0",
          "height": "100%",
          "left": "0",
          "overflow": "hidden",
          "position": "absolute",
          "top": "0",
          "width": "100%",
        }
      }
    />
    <div
      data-testid="publicIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-icon-2"
        focusable="false"
        id="publicIcon"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 17.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.21.21-1.79L9 15v1c0 1.1.9 2 2 2v1.93zm6.9-2.54c-.26-.81-1-1.39-1.9-1.39h-1v-3c0-.55-.45-1-1-1H8v-2h2c.55 0 1-.45 1-1V7h2c1.1 0 2-.9 2-2v-.41c2.93 1.19 5 4.06 5 7.41 0 2.08-.8 3.97-2.1 5.39z"
        />
      </svg>
    </div>
  </div>
</div>
`;
